torch/_dynamo/symbolic_convert.py

```diff
@@ -1941,6 +1941,26 @@ def RETURN_VALUE(self, inst):
         self.output.compile_subgraph(
             self, reason=GraphCompileReason("return_value", [self.frame_summary()])
         )
+        tos = self.stack[-1]
+        from .variables.tensor import NumpyTensorVariable
+
+        # This is a hack to handle NumpyTensorVariable before returning it. This actually calls Tensor.numpy() to
+        # convert tensor to ndarray. Note that this doesn't handle TupleVariable so if more than one return values
+        # and one of them is NumpyTensorVariable this doesn't convert it and the result will be tensor instead of
+        # ndarray.
```

Comment:
Why can't you do this properly with the reconstruct in the VariableTracker?

Reply:
@ezyang I think we should convert the fake_numpy_tensor back to numpy at ```RETURN_VALUE```. Look at this example:
```
def fn(x, y):
    a = x.numpy()
    b = y.numpy()
    torch._dynamo.graph_break()
    a = numpy.sin(a + 1)
    return a.T + b

x = torch.rand(2, 3)
y = torch.rand(3, 2)
print(fn(x, y))
```
If we do nothing, the return value is ```NumpyTensorVariable``` which is actually a tensor when switch back to python world. My suggestion here is to wrap these logics as a ```NumpyTensorVariable``` member method for better maintenance. 

> Why can't you do this properly with the reconstruct in the VariableTracker?

Do you mean to support graph break between many numpy ops(sth like in my example)? If yes, we should wrap the fake_numpy_tensor, compile the partial graph and get the numpy value when handling the graph break. Then during reconstruction, we directly load the numpy value and wrap back to  fake_numpy_tensor. Let me know if we are on the same page. Thanks.

Reply:
we talked about this briefly; while we do need logic in reconstruct, we need different logic for RETURN_VALUE, and Yanbo will figure out where it should go

Reply:
Moved to output_graph.py as well as tensor.py. Need to handle tuple variable later.